include/avif/avif.h

```diff
@@ -1414,15 +1414,31 @@ AVIF_API avifResult avifDecoderNthImageMaxExtent(const avifDecoder * decoder, ui
 // ---------------------------------------------------------------------------
 // avifEncoder
 
+struct avifEncoder;
 struct avifEncoderData;
 struct avifCodecSpecificOptions;
+struct avifEncoderCustomEncodeImageArgs;
 
 typedef struct avifScalingMode
 {
     avifFraction horizontal;
     avifFraction vertical;
 } avifScalingMode;
 
+// If enabled in avifEncoder, called for each coded image item (color, alpha, and gain map), grid cell,
+// and sequence (color, alpha).
+// Returns AVIF_RESULT_OK if it overrides the AV1 codec encoding pipeline for that element.
+// Returns AVIF_RESULT_NO_CONTENT if the AV1 codec encoding pipeline should be run for that element.
+// Returns an error otherwise.
+typedef avifResult (*avifEncoderCustomEncodeImageFunc)(struct avifEncoder * encoder,
+                                                       const avifImage * image,
+                                                       const struct avifEncoderCustomEncodeImageArgs * args);
+// Only called if avifEncoderCustomEncodeImageFunc returned AVIF_RESULT_OK.
+// Returns AVIF_RESULT_OK every time it outputs an AV1 sample.
+// Returns AVIF_RESULT_NO_IMAGES_REMAINING once all samples were output.
+// Returns an error otherwise.
+typedef avifResult (*avifEncoderCustomEncodeFinishFunc)(struct avifEncoder * encoder, avifROData * sample);
```

Comment:
How does this function know which image's samples to output?

Reply:
Done through `avifEncoderCustomEncodeImageItem`.